.test/run.sh

```diff
@@ -605,6 +605,87 @@ fi
 	killoff
 	cid=$master_host
 	killoff
+
+	;&
+	replication)
+
+	echo -e "Test: Replica container can be initialized with environment variables\n"
+
+	mariadb_replication_user='foo'
+	mariadb_replication_password='foo123'
+
+	cname="mariadbcontainer$RANDOM"
+	cid="$(
+		docker run --name "$cname"  \
+			--rm --publish 3306 \
+			-e MARIADB_REPLICATION_MODE="master" \
+			-e MARIADB_REPLICATION_USER="$mariadb_replication_user" \
+			-e MARIADB_REPLICATION_PASSWORD="$mariadb_replication_password" \
+			-e MARIADB_DATABASE=replcheck \
+			-e MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1
+			"$image" --server-id=3000 --log-bin --log-basename=my-mariadb
+	)"
+	waiting=${DOCKER_LIBRARY_START_TIMEOUT:-10}
+	echo "waiting to start..."
+	set +e +o pipefail +x
+	while [ "$waiting" -gt 0 ]
+	do
+		(( waiting-- ))
+		sleep 1
+		if ! docker exec -i "$cid" "$mariadb" -h localhost --protocol tcp -P 3306 -e 'select 1' 2>&1 | grep -F "Can't connect" > /dev/null
+		then
+			break
+		fi
+    done
+	set -eo pipefail -x
+	if [ "$waiting" -eq 0 ]
+	then
+		die 'timeout'
+	fi
+
+	grants=$(mariadbclient -u $mariadb_replication_user -p $mariadb_replication_password replcheck -e "SHOW GRANTS")
+
+	[ "${grants}" == "GRANT REPLICATION REPLICA ON *.* TO '$mariadb_replication_user'@'localhost'" ] || die "I wasn't created how I was expected"
+
+	# TODO: Create a Table in replcheck DB & Insert some rows in that to count.
```

Comment:
@grooverdan Do I really need to create a table in the created DB `replcheck` with some rows to perform data match between master & replica containers ?

Reply:
Is it problematic?

https://mariadb.com/kb/en/show-binlog-status/ to retreive the master position after the transactions and https://mariadb.com/kb/en/master_pos_wait/ to wait on the replica until they are committed.

Alternately, the gtid position - https://mariadb.com/kb/en/master_gtid_wait/.

Reply:
I was thinking to find a way to check replication is done successfully or not, So counting table data rows of master & replica was a way to confirm replication.